src/psyclone/domain/lfric/algorithm/psyir/lfric_alg_invoke_call.py

```diff
@@ -68,6 +68,21 @@ def _def_container_root_name(node):
         '''
         return f"{node.name}_psy"
 
+    def _def_routine_root_name(self):
+        '''
+        :returns: the proposed processed routine name for this invoke.
+        :rtype: str
+
+        '''
+        name = super()._def_routine_root_name()
+        if (len(self.children) == 1 and
+                isinstance(self.children[0], LFRicBuiltinFunctor)):
+            # By default the name of the kernel is added if there is
+            # only one functor. However we don't add this in LFRic if
+            # the functor is a builtin.
+            name = f"invoke_{self._index}"
```

Comment:
perhaps:

        return f"invoke_{self._index}"
    else:
        return super()._def_routine_root_name()

Reply:
Modified (but without the 'else')